src/commands/register.ts

```diff
@@ -118,40 +127,61 @@ export default class ComponentRegister extends Command {
     const project_name = `register.${resourceRefToNodeRef(component_spec.name)}.${tag}`;
     const compose_file = DockerComposeUtils.buildComposeFilepath(this.app.config.getConfigDir(), project_name);
 
-    await DockerComposeUtils.writeCompose(compose_file, yaml.dump(compose));
+    await DockerBuildXUtils.writeCompose(compose_file, yaml.dump(compose));
```

Comment:
This can still be `DockerComposeUtils` since it is a compose file. `BuildX` just happens to support compose file format. This also means we don't duplicate functionality.